drivers/ipvlan/ipvlan_network.go

```diff
@@ -0,0 +1,174 @@
+package ipvlan
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/libnetwork/driverapi"
+	"github.com/docker/libnetwork/netlabel"
+	"github.com/docker/libnetwork/options"
+	"github.com/docker/libnetwork/types"
+)
+
+// CreateNetwork the network for the specified driver type
+func (d *driver) CreateNetwork(id string, option map[string]interface{}, ipV4Data, ipV6Data []driverapi.IPAMData) error {
+	// parse and validate the config and bind to networkConfiguration
+	config, err := parseNetworkOptions(id, option)
+	if err != nil {
+		return err
+	}
+	if id == "" {
+		return fmt.Errorf("Network ID cannot be empty")
+	}
+	config.ID = id
+	err = config.processIPAM(id, ipV4Data, ipV6Data)
+	if err != nil {
+		return err
+	}
+	// verify the macvlan mode from -o ipvlan_mode option
+	switch config.IpvlanMode {
+	case "", modeL2:
+		// default to ipvlan L2 mode if -o ipvlan_mode is empty
+		config.IpvlanMode = modeL2
+	case modeL3:
+		config.IpvlanMode = modeL3
+	default:
+		return fmt.Errorf("requested ipvlan mode '%s' is not valid, 'l2' mode is the ipvlan driver default", config.IpvlanMode)
+	}
+	// user must specify a parent interface on the host to attach the container vif -o host_iface=eth0
+	if config.HostIface == "" {
+		return fmt.Errorf("%s requires an interface from the docker host to be specified (usage: -o host_iface=eth)", ipvlanType)
+	}
+	err = d.createNetwork(config)
+	if err != nil {
+		return err
+	}
+	// update persistent db with the new network
+	return d.storeUpdate(config)
+}
+
+// createNetwork is used by new network callbacks and persistent network cache
+func (d *driver) createNetwork(config *configuration) error {
+	// if the -o host_iface does not exist, attempt to parse a parent_iface.vlan_id
+	if ok := validateHostIface(config.HostIface); !ok {
+		// if the subinterface parent_iface.vlan_id checks do not pass, return err.
+		//  a valid example is eth0.10 for a parent iface: eth0 with a vlan id: 10
+		err := validateSubIface(config.HostIface)
+		if err != nil {
+			return err
+		}
+		// once the subinterface naming is validated create the link name.vlan_id
+		err = createVlanSubIface(config.HostIface)
+		if err != nil {
+			return err
+		}
+	}
+	n := &network{
+		id:        config.ID,
+		driver:    d,
+		endpoints: endpointTable{},
+		config:    config,
+	}
+	// update *network
+	d.addNetwork(n)
+	return nil
+}
+
+// DeleteNetwork the network for the specified driver type
+func (d *driver) DeleteNetwork(nid string) error {
+	if nid == "" {
+		return fmt.Errorf("network ID cannot be empty")
+	}
+	n := d.network(nid)
+	if n == nil {
+		return fmt.Errorf("could not find network with ID %s", nid)
+	}
+	// if the interface exists, only delete if it matches iface.vlan naming
+	if ok := validateHostIface(n.config.HostIface); ok {
```

Comment:
Im game, to clarify do you mean another `-o vlan_id=`. I initially did but there was some feedback on too many options. I tend to like being explicit since inferring can have conflicts. Currently if it exists it uses it to allow for custom naming but another option keeps things clean.